src/pint/models/binary_ell1.py

```diff
@@ -283,3 +273,14 @@ def validate(self):
             self.binary_instance.ds_func = self.binary_instance.delayS_H3_STIGMA_exact
             if self.STIGMA.quantity <= 0:
                 raise ValueError("STIGMA must be greater than zero.")
+        self.update_binary_object()
+
+    def validate(self):
```

Comment:
Should users expect a warning or an exception if they try to use `SINI` and `M2` with `ELL1H`? I would prefer an exception.

Reply:
It is in the Validate function. The changes are not showing everything. I can change it to an exception.

Reply:
It should be an exception now